File: 0011-container-with-most-water/0011-container-with-most-water.java
// Two Pointer Approach
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int maxArea = 0;
        while(left<right){
            int Area = Math.min(height[left],height[right])*(right-left);
            maxArea = Math.max(maxArea,Area);
            if(height[left]<height[right])
            left++;
            else
            right--;
        }
        return maxArea;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
    public:
        int maxArea(vector<int>& height) {
            int left = 0;
            int right = height.size()-1;
            int MaxArea = 0;
            while(left<right){
                int Area = min(height[left] , height[right])*(right-left);
                MaxArea = max(Area,MaxArea);
                if(height[left]<height[right]){
                    left++;
                }
                else{
                    right--;
                }
            }
            return MaxArea;
        }
    };